### PR TITLE
Totp part3

### PR DESCRIPTION
**Pull request by 0x0ptr**

### CODE REVIEW

File: source_code/main_mcu/src/LOGIC/logic_database.c [hunk 1]
```diff
@@ -813,6 +851,51 @@ RET_TYPE logic_database_add_credential_for_service(uint16_t service_addr, cust_c
     return ret_val;
 }
 
+/*! \fn     logic_database_add_TOTP_credential_for_service(uint16_t service_addr, cust_char_t* login, TOTPcredentials_t const *TOTPcreds)
```
Comment (limpkin) [inline source_code/main_mcu/src/LOGIC/logic_database.c:854]: ctr missing from doc :)

File: source_code/main_mcu/src/LOGIC/logic_database.c [hunk 2]
```diff
@@ -813,6 +851,51 @@ RET_TYPE logic_database_add_credential_for_service(uint16_t service_addr, cust_c
...
+RET_TYPE logic_database_add_TOTP_credential_for_service(uint16_t service_addr, cust_char_t* login, TOTPcredentials_t const *TOTPcreds, uint8_t *ctr)
+{
+    uint16_t storage_addr = NODE_ADDR_NULL;
+    child_cred_node_t temp_cnode;
+
+    /* Clear node */
+    memset((void*)&temp_cnode, 0, sizeof(temp_cnode));
+
+    /* Update fields that are required */
+    utils_strncpy(temp_cnode.login, login, sizeof(temp_cnode.login)/sizeof(cust_char_t));
```
Comment (limpkin) [inline source_code/main_mcu/src/LOGIC/logic_database.c:872]: MEMBER_ARRAY_SIZE macro is also an option :)

File: source_code/main_mcu/src/LOGIC/logic_database.c [hunk 3]
```diff
@@ -844,6 +927,30 @@ void logic_database_fetch_encrypted_password(uint16_t child_node_addr, uint8_t*
...
+    child_cred_node_t temp_cnode;
+
+    /* Read node, ownership checks and text fields sanitizing are done within */
+    nodemgmt_read_cred_child_node(child_node_addr, &temp_cnode);
+
+    /* Copy encrypted password */
+    memcpy(TOTPsecret, temp_cnode.TOTP.TOTPsecret, sizeof(temp_cnode.TOTP.TOTPsecret));
+
+    /* Copy secret (non-encryped) TOTPsecret length */
+    *TOTPsecretLen = temp_cnode.TOTP.TOTPsecretLen;
```
Comment (limpkin) [inline source_code/main_mcu/src/LOGIC/logic_database.c:948]: needs to be sanitized.

File: source_code/main_mcu/src/LOGIC/logic_encryption.c [hunk 4]
```diff
@@ -414,15 +414,17 @@ static uint32_t logic_encryption_sha1_truncate(uint8_t const *sha1)
     return truncated_value;
 }
 
-/*! \fn     logic_encryption_generate_totp(uint8_t *key, uint8_t key_len, uint8_t num_digits, cust_char_t *str, uint8_t str_len)
+/*! \fn     logic_encryption_generate_totp(uint8_t *key, uint8_t key_len, uint8_t num_digits, uint8_t timeStep, cust_char_t *str, uint8_t str_len)
```
Comment (limpkin) [inline source_code/main_mcu/src/LOGIC/logic_encryption.c:417]: different naming :)

**Comment by limpkin** [comment]
merging, can you make a tiny PR for the remaining tiny points? :)

File: source_code/main_mcu/src/LOGIC/logic_database.c (hunk 1 above)
Comment (0x0ptr) [reply source_code/main_mcu/src/LOGIC/logic_database.c:854]: :+1:

File: source_code/main_mcu/src/LOGIC/logic_database.c (hunk 2 above)
Comment (0x0ptr) [reply source_code/main_mcu/src/LOGIC/logic_database.c:872]: :+1:

File: source_code/main_mcu/src/LOGIC/logic_encryption.c (hunk 4 above)
Comment (0x0ptr) [reply source_code/main_mcu/src/LOGIC/logic_encryption.c:417]: :+1:

**Pull request merged**